reconcile/typed_queries/glitchtip_settings.py

```diff
@@ -1,18 +1,33 @@
+from pydantic import BaseModel
+
+from reconcile.gql_definitions.fragments.vault_secret import VaultSecret
 from reconcile.gql_definitions.glitchtip.glitchtip_settings import query
 from reconcile.utils import gql
 
 
-def get_glitchtip_settings(
-    read_timeout: int = 30, max_retries: int = 3, mail_domain: str = "redhat.com"
-) -> tuple[int, int, str]:
+class GlitchtipSettings(BaseModel):
+    read_timeout: int = 30
+    max_retries: int = 3
+    mail_domain: str = "redhat.com"
+    glitchtip_jira_bridge_alert_url: str = "https://gjb.devshift.net/api/v1/alert"
+    glitchtip_jira_bridge_token: VaultSecret | None
+
+
+def get_glitchtip_settings() -> GlitchtipSettings:
     """Returns Glitchtip Settings."""
     gqlapi = gql.get_api()
-    if _s := query(query_func=gqlapi.query).settings:
-        if _gs := _s[0].glitchtip:
-            if _gs.read_timeout is not None:
-                read_timeout = _gs.read_timeout
-            if _gs.max_retries is not None:
-                max_retries = _gs.max_retries
-            if _gs.mail_domain is not None:
-                mail_domain = _gs.mail_domain
-    return read_timeout, max_retries, mail_domain
+    settings = query(query_func=gqlapi.query).settings
+    ai_gs = settings[0].glitchtip if settings else None
+    gs = GlitchtipSettings()
+    if ai_gs:
+        if ai_gs.read_timeout:
+            gs.read_timeout = ai_gs.read_timeout
+        if ai_gs.max_retries:
+            gs.max_retries = ai_gs.max_retries
+        if ai_gs.mail_domain:
+            gs.mail_domain = ai_gs.mail_domain
+        if ai_gs.glitchtip_jira_bridge_alert_url:
+            gs.glitchtip_jira_bridge_alert_url = ai_gs.glitchtip_jira_bridge_alert_url
+        if ai_gs.glitchtip_jira_bridge_token:
+            gs.glitchtip_jira_bridge_token = ai_gs.glitchtip_jira_bridge_token
```

Comment:
Do you mean something like this?

```python
GlitchtipSettings(**ai_gs.dict(exclude_none=True))
```

Reply:
Unfortunately, this doesn't work because of the aliases in the `VaultSecret` GQL class :( and I don't want to copy that class.

Reply:
@geoberle [suggested](https://github.com/app-sre/qontract-schemas/pull/535#discussion_r1360853341) to move the settings to the instance object. I did that in [7f56daa](https://github.com/app-sre/qontract-reconcile/pull/3880/commits/7f56daa3784b36873998dcfae9e9ccf346b2e5a6), and now it's much cleaner :)

Reply:
Nice, one issue with that or https://github.com/app-sre/qontract-reconcile/pull/3880/commits/7f56daa3784b36873998dcfae9e9ccf346b2e5a6#diff-4ce1a595f5fdda9d51fb6cd5777e6c849c81c115924afc45030f00af91d42143R42-R43 is `0` will be override to `3` or `30`, is that expected?

Reply:
You have 🦅 👀. Changed in [31f3d38](https://github.com/app-sre/qontract-reconcile/pull/3880/commits/31f3d389306f2ce5c5a570ae2b7b698c42fca526)